firefox_ui_tests/remote/security/test_safe_browsing_notification.py

```diff
@@ -0,0 +1,120 @@
+
+from marionette_driver import (
+    By,
+    expected
+)
+
+from firefox_ui_harness.testcase import FirefoxTestCase
+
+
+class TestSafeBrowsingNotificationBar(FirefoxTestCase):
+
+    def setUp(self):
+        FirefoxTestCase.setUp(self)
+
+        self.test_data = [
+            # Phishing URL info
+            {
+                'button_label': 'safebrowsing.notAForgeryButton.label',
+                'report_page': 'www.google.com/safebrowsing/report_error',
+                'unsafe_page': 'https://www.itisatrap.org/firefox/its-a-trap.html'
+            },
+            # Malware URL object
+            {
+                'button_label': 'safebrowsing.notAnAttackButton.label',
+                'report_page': 'www.stopbadware.org',
+                'unsafe_page': 'https://www.itisatrap.org/firefox/its-an-attack.html'
+            }
+        ]
+
+        self.prefs.set_pref('browser.safebrowsing.enabled', True)
+        self.prefs.set_pref('browser.safebrowsing.malware.enabled', True)
+
+        # workaround for bug 1084289; in smaller windows, clipping causes X button to disapper
+        self.marionette.maximize_window()
+
+        # Give the browser a little time, because SafeBrowsing.jsm takes a
+        # while between start up and adding the example urls to the db.
+        # hg.mozilla.org/mozilla-central/file/46aebcd9481e/browser/base/content/browser.js#l1194
+        time.sleep(3)
+
+    def tearDown(self):
+        try:
+            self.browser.tabbar.close_all_tabs([self.browser.tabbar.tabs[0]])
+        finally:
+            FirefoxTestCase.tearDown(self)
+
+    def test_notification_bar(self):
+        with self.marionette.using_context('content'):
+            for item in self.test_data:
+                label = item['button_label']
+                report_page, unsafe_page = item['report_page'], item['unsafe_page']
+
+                # Navigate to the unsafe page
+                # Check "ignore warning" link then notification bar's "not badware" button
+                self.marionette.navigate(unsafe_page)
+                self.check_ignore_warning_button(unsafe_page)
+                self.check_not_badware_button(label, report_page)
+
+                # Return to the unsafe page
+                # Check "ignore warning" link then notification bar's "get me out" button
+                self.marionette.navigate(unsafe_page)
+                self.check_ignore_warning_button(unsafe_page)
+                self.check_get_me_out_of_here_button()
+
+                # Return to the unsafe page
+                # Check "ignore warning" link then notification bar's "X" button
+                self.marionette.navigate(unsafe_page)
+                self.check_ignore_warning_button(unsafe_page)
+                self.check_x_button()
+
+    def check_ignore_warning_button(self, unsafe_page):
+        button = self.marionette.find_element(By.ID, 'ignoreWarningButton')
+
+        # Wait for the DOM to receive events
+        time.sleep(1)
```

Comment:
I may have asked this earlier but why do we need this is it is not an about:error page? If we do I would like to see it right next to the call to `navigate()`.